packages/playback-core/src/index.ts

```diff
@@ -397,7 +399,16 @@ export const loadMedia = (
     switch (mediaEl.preload) {
       case 'none':
         // when preload is none, load the source on first play
-        mediaEl.addEventListener('play', () => hls.loadSource(src), { once: true });
+        const loadSrcOnPlay = () => hls.loadSource(src);
+        mediaEl.addEventListener('play', loadSrcOnPlay, { once: true });
+
+        mediaEl.addEventListener(
```

Comment:
given that we rely on this pattern throughout to ensure proper reference cleanup, we may want to just make a function to do this for us. defer to others on priority.

Reply:
yeah, having a "onEventPerSource", or something like that, function would be nice.